### ISSUE TITLE
How to achieve nested element styling?

### ISSUE THREAD

**Issue by FarhadG** [rofrischmann/react-look#294]
I've looked through the docs and demos, however, I don't see what's the prescribed way of handling nested elements styling. For example, if I have a `<button> element with a nested `<span>`, how to have the span element also update when the user `hovers` over the parent `<button>` element?

```
<button className={styles.button}>
  <span className={styles.label}>Hello</span>
</button>
```

**Comment by rofrischmann** [comment]
Well you're example pretty well shows how to do nested styling right? Or what do you mean by update when the user hovers? How do you trigger style changes on hover?

**Comment by FarhadG** [comment]
Yes, that's exactly right. For example, when the user hovers over the `button` (which is larger than the span), I'd like the `label` to also be treated differently. By setting the `:hover` state on the label, it works, however, you must actually hover over it, hence why I was hoping the parent-child relationship within CSS could somehow carry over here: e.g. `span.className:hover label.className { ... }`

**Comment by rofrischmann** [comment]
Oh I see. Sadly that's something we do not officially support. But there's a simple hack to achieve this. You could go with sth. like:
```javascript
styles = StyleSheet.create({
  button: {
    ':hover > span' : {
       ...
    }
  }
})
```

**Comment by FarhadG** [comment]
Thank you for the quick response, @rofrischmann. I'm loving this library and was thinking of using it in a bigger project, however, as I've been reading through the issues, it seems like you're not going to be actively maintaining this and, instead, working on another library? My concern is, basically, not wanting to  advise this library be used and, then, it no longer can be maintained, etc. Thoughts?

Appreciate the help and solid effort...

**Comment by rofrischmann** [comment]
I would recommend using [Fela](https://github.com/rofrischmann/fela) if you build from scratch. I will release 1.0.0 next week and its already running solid in production.
Yet if you think Look is better than Fela, there's no problem using it. I am using it in several projects myself and everything works just fine. You're right I am searching for some help maintaining it as I use quite a lot of my free time to handle all the other packages, but still I do not "stop" updating Look.

Fela's API actually differs from Looks in some key points, but the idea stays the same somehow. Yet I believe it is not hard to get started with Fela. It also is capable of the same features I guess.

**Comment by FarhadG** [comment]
Thank you for the detailed response, @rofrischmann. I'll start looking into Fela; I'm curious does it offer the nested element styling as I mentioned? This is something that we find ourselves using quite a bit and nested the style selector with `:hover > span` would give us problems, as we'd generally want a way to select a class within the sub selection (e.g. `:hover .className`).

**Comment by rofrischmann** [comment]
Well sadly also Fela does not support that by design. It was an active decision to leave out any selector combinators as it increases both specificity problems as well as complexity. I am curious why you actually need it as I believe there are for sure ways to achieve the same goal without those selectors =) 
You could implement onHover and pass the state to an inner Component e.g. but that ofc is not a solid solution either.

**Comment by FarhadG** [comment]
Yeah, I agree that composite selectors are not really the way to go about it with the whole `css-in-approach` but I was thinking that it would be cool to do the following:

```js
...
button: {
  `:hover`: { 
     color: 'red',
     // solution 1 - something along the lines of being able to capture the label element ... ehh...
     'element=label': { 
       color: 'red'
     }
   }
},
label: {
  color: 'black':
  // solution 2 - probably better
  ':hover': (props, state, context) => //somehow refer to `button` is `:hover` is true, return `value`
}
...
```
